### ISSUE TITLE
unable to uninstall install start debian image when sudoers list file gets corrupted

### ISSUE THREAD

**Issue by jakegrabowski** [microsoft/WSL#7444]
### Windows Build Number

21H1 19043.1237

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.16

### Distro Version

Debian

### Repro Steps

Description:
yesterday I edited sudoers list apparently i corrupted it and today I get "Unable to sudo: PAM authentication error: Module is unknown" error so I tried to fix and then reset my debian but this caused that debian image dissapeard from programs and features and Im unable to download it from Microsoft store bacause I get error now. Other images still work.

Steps to reproduce:
1. sudo visudo /etc/sudoers
2. corrupt file
3. restart your distribution - I restarted computer
4. now whenever you will use sudo you should get error "Unable to sudo: PAM authentication error: Module is unknown"
5. in windows programs & features select debian advanced and click fix and then click reset

### Expected Behavior

Expected was my Debian is reinstalled

### Actual Behavior

Above Steps Caused:

1. Debian image dissapeared from programs & features
2. Im unable to install Debian from Microsoft store
3. wsl -l do not show Debian on the list

**Comment by craigloewen-msft** [comment]
@jakegrabowski what error are you seeing when you try to install Debian? 
Can you uninstall the distribution in Debian and then reinstall it?

If you open the "Debian" app what happens?

**Comment by jakegrabowski** [comment]
Hello Craig
Today I started computer and issue is gone. I was able today to install Debian from Microsoft store.

Yesterdays error said - an error occured - nothing really descriptive.

Today I did also additional investigation and tried to break Debian again using my steps so I edited "sudo visudo /etc/sudoers" and damaged this configuration file it in a way that root and users from sudoers group was not able to perform any sudo operation.
Then in programs & features I selected Debian advanced and click fix and then click reset (this time I did this twice).

For some short period of time it seemed that I reproduced issue but then I was again able to run Debian. Not like yesterday when I spent 30 minutes of time to figure out that something is wrong.

It seems that yesterday resetting Debian from Programs & Features made reset script stuck at some point, for some reason, but today I am unable to reproduce the issue.
